### DISCUSSION TITLE
Instability in my 74x4040 implementation

### DISCUSSION THREAD

**Discussion by chyvonomys** [hneemann/Digital#1008] [Q&A]
Hi. Library was lacking this chip so I tried implementing it using JK triggers (a) and three 74x161 (b). Variant (b) is not stable: QA is randomly 1/0 when I hit play button. Why is it so? How do I fix this?
![74x4040](https://user-images.githubusercontent.com/2478891/181933421-af098208-763c-47eb-98b7-54e927a66706.PNG)

**Comment by hneemann** [comment]
I suspect that there is a race condition between the two inverters. 
If they do not switch at the same time, it may happen that 1 or 0 is shown at QA, depending on which inverter is faster.
Either there is a CLR first and then a CLK or vice versa.

**Comment by mengstr** [comment]
The variant with 74161 is not a good representation of the 4040 anyways. The 4040 is a ripple counter and the 7416x-series are synchronous counters which makes them behave differently.
